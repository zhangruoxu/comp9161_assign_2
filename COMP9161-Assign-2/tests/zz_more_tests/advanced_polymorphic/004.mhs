main = 
  let g = letfun f x = (letfun h z = x, x+1);
  in g;
