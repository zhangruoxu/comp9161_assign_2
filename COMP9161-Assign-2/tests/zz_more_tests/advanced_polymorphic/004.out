main :: forall k. (Int -> (k -> Int) * Int) = let g :: forall e. (Int -> (e -> Int) * Int) = (letfun f :: (Int -> (e -> Int) * Int) x = ((letfun h :: (e -> Int) z = x), (+) x 1));
                                              in g;
